crates/re_space_view_spatial/src/space_view_3d.rs

```diff
@@ -84,46 +83,40 @@ impl SpaceViewClass for SpatialSpaceView3D {
-                    let mut entities_under_pinholes = IntSet::<EntityPath>::default();
-
-                    for (child_origin, connection) in &primary_space.child_spaces {
-                        if connection.is_connected_pinhole() {
-                            let Some(child_space) =
-                                topo.subspace_for_subspace_origin(child_origin.hash())
-                            else {
-                                // Should never happen, implies that a child space is not in the list of subspaces.
-                                continue;
-                            };
-
-                            // Entities _at_ pinholes are a special case: we display both 3d and 2d visualizers for them.
-                            entities_in_main_3d_space.insert(child_space.origin.clone());
-                            entities_under_pinholes.extend(child_space.entities.iter().cloned());
-                        }
-                    }
+                };
+            }
 
-                    VisualizableFilterContext3D {
-                        entities_in_main_3d_space,
-                        entities_under_pinholes,
-                    }
+            // All entities in the 3d space are visualizable + everything under pinholes.
+            let mut entities_in_main_3d_space = primary_space.entities.clone();
+            let mut entities_under_pinholes = IntSet::<EntityPath>::default();
+
+            for child_origin in &primary_space.child_spaces {
+                let Some(child_space) = topo.subspace_for_subspace_origin(child_origin.hash())
+                else {
+                    // Should never happen, implies that a child space is not in the list of subspaces.
+                    continue;
+                };
+
+                if child_space
+                    .connection_to_parent
+                    .contains(SubSpaceConnectionFlags::Pinhole)
+                {
+                    // Note that for this the connection to the parent is allowed to contain the disconnected flag.
+                    // Entities _at_ pinholes are a special case: we display both 3d and 2d visualizers for them.
+                    entities_in_main_3d_space.insert(child_space.origin.clone());
+                    entities_under_pinholes.extend(child_space.entities.iter().cloned());
                 }
+            }
 
-                SubSpaceDimensionality::TwoD => {
-                    // If this is 2D space, only display the origin entity itself.
-                    // Everything else we have to assume requires some form of transformation.
-                    VisualizableFilterContext3D {
-                        entities_in_main_3d_space: std::iter::once(space_origin.clone()).collect(),
-                        entities_under_pinholes: Default::default(),
-                    }
-                }
+            VisualizableFilterContext3D {
+                entities_in_main_3d_space,
```

Comment:
this should recursively collect all entities that are in child spaces that are not excluding 3D spaces.
this recursion needs to go both down the children and up the parent.

Similarly we also need to collect more pinholes!